src/Client.php

```diff
@@ -63,6 +68,8 @@ public function __construct(array $config, $clientClass = GuzzleClient::class)
         $this->version = urlencode($config['version']);
         $this->client_id = $config['client_id'];
         $this->client_secret = $config['client_secret'];
+
+        $this->resultsAsArray = isset($config['results_as_array']) ? (bool) $config['results_as_array'] : false;
```

Comment:
we don't need to check if it's set if you add it to the first statement:
```
$config += [
    'base_uri'       => 'https://api.classy.org',
    'check_ssl_cert' => true,
    'results_as_array' => false
];
```

Reply:
The issue is that if you pass, for example ['results_as_array' => []], and without a type check/cast you'll get a fatal error (if strict type is on) due to the mismatched types (otherwise you'll get null from the decode).

https://3v4l.org/TnYqv
https://3v4l.org/mQKiN

Reply:
Is this really something that we should support? PHP isn't my primary language, but afaik this is something a linter would be good for, anybody using this library who passes a non-boolean in for this config param would know immediately that they made a mistake during development.